spark-commons/src/main/scala/za/co/absa/spark/commons/implicits/DataTypeImplicits.scala

```diff
@@ -23,6 +23,16 @@ import za.co.absa.spark.commons.implicits.StructTypeImplicits.StructTypeEnhancem
 object DataTypeImplicits {
 
   implicit class DataTypeEnhancements(val dt: DataType) extends AnyVal {
+    /**
+     * Wraps an array of StructField with StructType
+     *
+     * @param fieldSchema is the actual schema to be wrapped
+     * @return StructType containing the StructFields
+     */
+    def wrapStructFiledWithStructType(fieldSchema: Option[DataType], errorColumnName: String): Boolean = {
+      val st = StructType(Seq(StructField(errorColumnName, fieldSchema.head)))
+      isEquivalentDataType(st)
+    }
```

Comment:
Couple of things wrong with this function (not mentioning it's not needed anymore, so please remove 😄 )
* Function name does not align with the result type (from name one would expect to get back a `StructField`
* `errorColumnName` parameter name is oddly specific to the place where the function was used. These are general functions, they should have a general use, and therefore the parameter names hint its reason in the function.
* `fieldSchema.head` relies on that the `Option` is non-empty. You should never assume that.
*